sql/src/main/java/org/opensearch/sql/sql/parser/AstExpressionBuilder.java

```diff
@@ -430,12 +430,13 @@ private UnresolvedExpression visitConstantFunction(String functionName,
   }
 
   private QualifiedName visitIdentifiers(List<IdentContext> identifiers) {
+    Boolean isMetadataField = identifiers.stream().filter(id -> id.metadataField() != null).findFirst().isPresent();
```

Comment:
Could be there a mix of different types of identifiers - some of them are metadata and some of them are not?